allensdk/brain_observatory/behavior/data_objects/stimuli/presentations.py

```diff
@@ -71,15 +71,72 @@ def __init__(
         if column_list is not None:
             presentations = presentations[column_list]
         if sort_columns:
-            presentations = presentations[sorted(presentations.columns)]
+            presentations = self._enforce_column_order(presentations)
         presentations = presentations.reset_index(drop=True)
+        presentations = self._enforce_int_typing(presentations)
         presentations.index = pd.Index(
             range(presentations.shape[0]),
             name="stimulus_presentations_id",
             dtype="int",
         )
+
         super().__init__(name="presentations", value=presentations)
 
+    def _enforce_column_order(
```

Comment:
this is extremely similar to some other code you wrote. please generalize it so there is no duplication